### PR TITLE
stream: skip try_select check for incoming streams

### PR DESCRIPTION
**Pull request by indutny**
This check is redundant for incoming TCP connection, and we're just
loosing performance on it. Skip it in the cases, where we already know
that fd works fine with kqueue.

/cc @bnoordhuis @piscisaureus

### CODE REVIEW

**Comment by bnoordhuis** [comment]
This is the wrong approach. The only place where 'exotic' file descriptors come in is in uv_pipe_open() and uv_tty_init() so that's where the 'is this a file descriptor that works with kqueue?' checks should be.

**Comment by indutny** [comment]
That's not really true statement, stream might be created using existing fd.

**Comment by bnoordhuis** [comment]
Yes, in those two functions I mentioned. There's also uv_tcp_open() but the implicit contract is that you pass it a socket so that one doesn't need special handling. That about covers it, doesn't it?

File: src/unix/pipe.c [hunk 1]
```diff
@@ -155,6 +155,16 @@ void uv__pipe_close(uv_pipe_t* handle) {
 
 
 int uv_pipe_open(uv_pipe_t* handle, uv_file fd) {
+#if defined(__APPLE__)
+  {
+    int r;
+
+    r = uv__stream_try_select((uv_stream_t*) handle, &fd);
```
Comment (bnoordhuis) [inline src/unix/pipe.c]:
``` c
if (uv__stream_try_select((uv_stream_t*) handle, &fd))
  return -1;
```

?

**Comment by indutny** [comment]
All nits fixed, anything else?

**Comment by bnoordhuis** [comment]
Nice work, Fedor. Landed in b45a74f.

**Pull request closed without merging**